From 9f1c7313073898d0c2a2fd321396d49a03f242e2 Mon Sep 17 00:00:00 2001
From: abhishek262.kumar <Abhishek262.Kumar@ril.com>
Date: Mon, 12 Jun 2023 17:42:57 +0530
Subject: [PATCH] [PATCH] [Skyworth]Refactor reset key code including bootloader,  kernel, recovery

Change-Id: I2936bfff8ecc362dc638f4988cd520e511d81bdf
---

diff --git a/external/recovery/recovery/amlogic_ui.cpp b/external/recovery/recovery/amlogic_ui.cpp
index 7c72b93..4e470f6 100644
--- a/external/recovery/recovery/amlogic_ui.cpp
+++ b/external/recovery/recovery/amlogic_ui.cpp
@@ -62,12 +62,14 @@
     { "select", KEY_ENTER, {KEY_ENTER, KEY_TAB, KEY_BACK, -1, -1, -1} },
     { "down", KEY_DOWN, {KEY_DOWN, KEY_VOLUMEDOWN, KEY_PAGEDOWN, -1, -1, -1} },
     { "up", KEY_UP, {KEY_UP, KEY_VOLUMEUP, KEY_PAGEUP, -1, -1, -1} },
+    { "reset", 249, {249, -1, -1, -1, -1, -1} },
 };
 
 CtrlInfo_t g_ctrlinfo[3] = {
     { "select", KEY_ENTER },
     { "down", KEY_DOWN },
     { "up", KEY_UP },
+    { "reset", 249 },
 };
 
 static KeyMapItem_t g_presupposed_keymap[] = {
diff --git a/external/recovery/recovery/amlogic_ui.h b/external/recovery/recovery/amlogic_ui.h
index 94e0906..2a7bab8 100644
--- a/external/recovery/recovery/amlogic_ui.h
+++ b/external/recovery/recovery/amlogic_ui.h
@@ -13,8 +13,8 @@
 #include <linux/input.h>
 #include <string>
 
-#define NUM_CTRLINFO 3
-#define NUM_DEFAULT_KEY_MAP 3
+#define NUM_CTRLINFO 4
+#define NUM_DEFAULT_KEY_MAP 4
 
 struct KeyMapItem_t {
     const char* type;
