From 970354fd19f7ec8d96ff6acc0d1c97675701ab1f Mon Sep 17 00:00:00 2001
From: abhishek262.kumar <Abhishek262.Kumar@ril.com>
Date: Mon, 12 Jun 2023 17:18:30 +0530
Subject: [PATCH] [PATCH] [Skyworth]Refactor reset key code including bootloader,  kernel, recovery

Change-Id: Id63c9a1df09d539c2f3b2b68ca7d6d4d6649ee1e
---

diff --git a/recovery_ui/include/recovery_ui/ui.h b/recovery_ui/include/recovery_ui/ui.h
index 071efce..08ec1d7 100644
--- a/recovery_ui/include/recovery_ui/ui.h
+++ b/recovery_ui/include/recovery_ui/ui.h
@@ -192,8 +192,6 @@
     return key_interrupted_;
   }
 
-  void ProcessKey(int key_code, int updown);
-
  protected:
   void EnqueueKey(int key_code);
 
@@ -225,6 +223,7 @@
   void OnKeyDetected(int key_code);
   void OnTouchDetected(int dx, int dy);
   int OnInputEvent(int fd, uint32_t epevents);
+  void ProcessKey(int key_code, int updown);
   void TimeKey(int key_code, int count);
 
   bool IsUsbConnected();
@@ -267,8 +266,6 @@
   std::thread input_thread_;
   std::atomic<bool> input_thread_stopped_{ false };
 
-  std::thread reset_gpio_thread_;
-
   ScreensaverState screensaver_state_;
 
   // The following two contain the absolute values computed from brightness_normal_ and
diff --git a/recovery_ui/ui.cpp b/recovery_ui/ui.cpp
index 7445659..c99cc15 100644
--- a/recovery_ui/ui.cpp
+++ b/recovery_ui/ui.cpp
@@ -139,39 +139,10 @@
   return true;
 }
 
-static void* ResetGpioThread(void* recovery_ui) {
-  RecoveryUI* ui = static_cast<RecoveryUI*>(recovery_ui);
-  char prev;
-  while(true) {
-    char ch;
-    int fd = open("/sys/devices/platform/reset_button/value", O_RDONLY);
-    if(read(fd,&ch,1) == 1) {
-      if (ch == '0' && ch != prev) {
-        prev = ch;
-        ui->ProcessKey(KEY_F13, 1);
-        LOG(WARNING) << " disabled";
-        printf("cys ResetGpioThread disabled");
-      } else if (ch == '1' && ch != prev) {
-        prev = ch;
-        ui->ProcessKey(KEY_F13, 0);
-        LOG(WARNING) << "enabled";
-        printf("cys ResetGpioThread enabled");
-      }
-    }
-    close(fd);
-    usleep(50000);
-  }
-}
-
 bool RecoveryUI::Init(const std::string& /* locale */) {
   ev_init(std::bind(&RecoveryUI::OnInputEvent, this, std::placeholders::_1, std::placeholders::_2),
           touch_screen_allowed_);
 
-  reset_gpio_thread_ = std::thread([this]() {
-    printf("ResetGpioThread called");
-    ResetGpioThread(this);
-  });
-
   ev_iterate_available_keys(std::bind(&RecoveryUI::OnKeyDetected, this, std::placeholders::_1));
 
   if (touch_screen_allowed_) {
@@ -601,7 +572,7 @@
     }
   }
 
-  if (key == KEY_F13) {
+  if (key == 249) {
     if (is_long_press && !IsTextVisible()) {
       return TOGGLE;
     }
