From 484a7b656828c51d3c71c81b52f7076ca3099bcc Mon Sep 17 00:00:00 2001
From: abhishek262.kumar <Abhishek262.Kumar@ril.com>
Date: Mon, 12 Jun 2023 17:23:23 +0530
Subject: [PATCH] [PATCH] [Skyworth]Refactor reset key code including bootloader,  kernel, recovery

Change-Id: I8bbe6c1597c596d2f5add0b8e7e7e2ec0863ee33
---

diff --git a/products/mbox/Vendor_0001_Product_0001.kl b/products/mbox/Vendor_0001_Product_0001.kl
index 52f742f..040bd5c 100644
--- a/products/mbox/Vendor_0001_Product_0001.kl
+++ b/products/mbox/Vendor_0001_Product_0001.kl
@@ -37,6 +37,8 @@
 key 109   VOLUME_DOWN
 key 212   CAMERA
 
+key 249   HARD_RESET
+
 #key 16    Q
 #key 17    W
 #key 18    E
diff --git a/products/mbox/Vendor_0001_Product_0002.kl b/products/mbox/Vendor_0001_Product_0002.kl
index 0b790b4..dd25bce 100644
--- a/products/mbox/Vendor_0001_Product_0002.kl
+++ b/products/mbox/Vendor_0001_Product_0002.kl
@@ -121,3 +121,5 @@
 
 key 600    PAIRING
 key 133    STEM_1
+
+key 249    HARD_RESET
diff --git a/products/mbox/sc2/recovery/recovery.kl b/products/mbox/sc2/recovery/recovery.kl
index aab4b41..6e36f56 100644
--- a/products/mbox/sc2/recovery/recovery.kl
+++ b/products/mbox/sc2/recovery/recovery.kl
@@ -18,4 +18,5 @@
 select		97		28		15		158
 down		108		114		109
 up		103		104		115
+reset		249
 back_door	128		26		118
diff --git a/sepolicy/file.te b/sepolicy/file.te
index b63e168..35154b3 100644
--- a/sepolicy/file.te
+++ b/sepolicy/file.te
@@ -108,5 +108,3 @@
 # /dev/aml_dsm
 type aml_dsm_device, fs_type,sysfs_type;
 
-# skyworth reset button
-type sk_reset, fs_type, sysfs_type;
diff --git a/sepolicy/file_contexts b/sepolicy/file_contexts
index a76d8a5..f22101c 100644
--- a/sepolicy/file_contexts
+++ b/sepolicy/file_contexts
@@ -523,5 +523,3 @@
 /sys/class/tee_info(/.*)?					      u:object_r:sysfs_tee:s0
 /dev/aml_dsm                                      u:object_r:aml_dsm_device:s0
 
-# skyworth reset button
-/sys/devices/platform/reset_button/value u:object_r:sk_reset:s0
diff --git a/sepolicy/recovery.te b/sepolicy/recovery.te
index 7e68b39..4d6b744 100644
--- a/sepolicy/recovery.te
+++ b/sepolicy/recovery.te
@@ -160,7 +160,3 @@
 allow recovery exfat:dir { open read search };
 allow recovery exfat:file { getattr open read };
 
-# skyworth reset button
-allow init sk_reset:file rw_file_perms;
-allow shell sk_reset:file rw_file_perms;
-allow recovery sk_reset:file rw_file_perms;
