From 86fea84a20e8acb1f5bcc411effc5e2b90a90429 Mon Sep 17 00:00:00 2001
From: abhishek262.kumar <Abhishek262.Kumar@ril.com>
Date: Mon, 12 Jun 2023 17:21:27 +0530
Subject: [PATCH] [PATCH] [Skyworth]Refactor reset key code including bootloader,  kernel, recovery

Change-Id: I4fa65aa653e32e67c0e690682f347fb0c886843b
---

diff --git a/arch/arm64/boot/dts/amlogic/sc2_s905x4_ah212_drm.dts b/arch/arm64/boot/dts/amlogic/sc2_s905x4_ah212_drm.dts
index 7f7ff98..9b0d128 100644
--- a/arch/arm64/boot/dts/amlogic/sc2_s905x4_ah212_drm.dts
+++ b/arch/arm64/boot/dts/amlogic/sc2_s905x4_ah212_drm.dts
@@ -383,12 +383,11 @@
 		compatible = "amlogic, gpio_keypad";
 		status = "okay";
 		scan_period = <20>;
-		key_num = <2>;
-		key_name = "bluetooth", "mute";
-		key_code = <600 SW_MUTE_DEVICE>;
-		key_type = <EV_KEY EV_SW>;
-		key-gpios = <&gpio  GPIOD_3  GPIO_ACTIVE_HIGH
-			     &gpio  GPIOE_2  GPIO_ACTIVE_HIGH>;
+		key_num = <1>;
+		key_name = "reset";
+		key_code = <249>;
+		key_type = <EV_KEY>;
+		key-gpios = <&gpio  GPIOA_15  GPIO_ACTIVE_HIGH>;
 		detect_mode = <0>;/*0:polling mode, 1:irq mode*/
 	};
 
@@ -784,7 +783,7 @@
 };
 
 &i2c3 {
-	status = "okay";
+	status = "disabled";
 	pinctrl-names="default";
 	pinctrl-0=<&i2c3_pins2>;
 	clock-frequency = <300000>; /* default 100k */
@@ -1264,8 +1263,8 @@
 
 	pdmin: pdmin {
 		mux { /* GPIOC_0, GPIOC_4 */
-			groups = "pdm_din0_c",
-				"pdm_dclk_c";
+			groups = "pdm_din0_z",
+				"pdm_dclk_z";
 			function = "pdm";
 		};
 	};
diff --git a/drivers/amlogic/input/keyboard/gpio_keypad.c b/drivers/amlogic/input/keyboard/gpio_keypad.c
index 4d5817e..3201f0c 100644
--- a/drivers/amlogic/input/keyboard/gpio_keypad.c
+++ b/drivers/amlogic/input/keyboard/gpio_keypad.c
@@ -137,8 +137,30 @@
 
 static CLASS_ATTR_RO(table);
 
+static ssize_t reset_show(struct class *cls, struct class_attribute *attr,
+		char *buf)
+{
+	struct gpio_keypad *keypad = container_of(cls,
+	struct gpio_keypad, kp_class);
+	int i;
+	int len = 0;
+	char *str_reset = "reset";
+
+	for (i = 0; i < keypad->key_size; i++)
+		if (!strcmp(keypad->key[i].name, str_reset)) {
+			len = sprintf(buf, "%d\n",
+				keypad->key[i].current_status);
+			break;
+		}
+
+	return len;
+}
+
+static CLASS_ATTR_RO(reset);
+
 static struct attribute *meson_gpiokey_attrs[] = {
 	&class_attr_table.attr,
+	&class_attr_reset.attr,
 	NULL
 };
 
